pkg/kubelet/pod_workers.go

```diff
@@ -298,6 +298,9 @@ func killPodNow(podWorkers PodWorkers, recorder record.EventRecorder) eviction.K
 		} else if pod.Spec.TerminationGracePeriodSeconds != nil {
 			gracePeriod = *pod.Spec.TerminationGracePeriodSeconds
 		}
+		if gracePeriod < 0 {
+			gracePeriod = 0
+		}
```

Comment:
Note there is similar logic in `pkg/kubelet/kuberuntime/kuberuntime_container.go` (no I don't know why there are two code paths) that would potentially also need an update. I ran into this because the `terminationGracePeriodSeconds` is used there when terminating on e.g. a failed probe.

https://github.com/kubernetes/kubernetes/blob/1fb27c4b5d6c9d8f508c7b4bbcf96244d6b65449/pkg/kubelet/kuberuntime/kuberuntime_container.go#L612-L619